NetBSD cross compiling

matthewbauer opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

matthewbauer committed:
netbsd: add cross target

Ericson2314 reviewed (approved):
Awesome!!

Ericson2314 commented:
Wooo! Staging and then it's a-go? We should get this on some their mailing list or whatever. It should be a great way to draw in more people to Nix.

matthewbauer committed:
bintools-wrapper: fix unknown emulation error message

Using the + operator with throw doesn’t seems to work properly. You
need to use antiquotes here to get the targetPlatform config included.

matthewbauer commented:
To avoid breaking anything I might wait until branching off for 19.03

matthewbauer commented:
I also am trying to upstream the GCC patch here:

https://gcc.gnu.org/ml/gcc-patches/2019-02/msg00198.html

Ericson2314 commented:
Good luck with the upstream!

Mic92 commented:
Hopefully easier then getting stuff into the LLVM.

vcunat commented on pkgs/top-level/all-packages.nix:
```diff
@@ -23258,8 +23260,10 @@ in
 
-  inherit (recurseIntoAttrs (callPackages ../os-specific/bsd { }))
-          netbsd;
+  netbsd = callPackages ../os-specific/bsd/netbsd {};
```
This removed `recurseIntoAttrs` and thus the packages aren't built by Hydra anymore: https://hydra.nixos.org/eval/1512985?filter=&compare=1512944&full=#tabs-removed  Was that intentional?

vcunat commented:
The bigger problem is that netbsd-make won't build on Darwin anymore, breaking lots of packages, e.g. https://hydra.nixos.org/job/nixpkgs/trunk/getent.x86_64-darwin

vcunat commented:
Well, `netbsd.make` won't build even on `x86_64-linux` on current nixpkgs master.

matthewbauer commented:
Very weird. I’ll look into it
